src/Lykke.Service.MicrographCache.Services/Extensions/CacheExt.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Threading.Tasks;
+using Common;
+using Microsoft.Extensions.Caching.Distributed;
+
+namespace Lykke.Service.MicrographCache.Services.Extensions
+{
+    public static class CacheExt
+    {
+        public static async Task<T> TryGetFromCache<T>(this IDistributedCache cache, string key,
+            Func<Task<T>> getRecordFunc, TimeSpan expiration)
+        {
+            var record = await TryGetRecordFromCache<T>(cache, key);
+
+            if (record == null)
+            {
+                record = await getRecordFunc();
+                await TryUpdateRecordInCache(cache, key, record, expiration);
+            }
+
+            return record;
+        }
+
+        private static async Task<T> TryGetRecordFromCache<T>(IDistributedCache cache, string key)
+        {
+            string value = await cache.GetStringAsync(key);
+
+            if (value != null)
+            {
+                return value.DeserializeJson<T>();
```

Comment:
I would use something lightweight like MessagePack or NetJSON instead of Json.Net

Reply:
why? cache holds just 20 doubles and updates it once per 1 min

Reply:
ok